### DISCUSSION TITLE
SHORT DESCRIPTION OF THE PROBLEM YOU ARE REPORTING

### DISCUSSION THREAD

**Discussion by wizardion** [photoprism/photoprism#2534] [Q&A]
#### 1. What is not working as documented?

It looks like that search by the keyword `original` doesn't work as expected from the docs (https://docs.photoprism.app/user-guide/organize/search/), on the demo website the existing file `demo/20020312-190154-Cab-2002.jpg` is not found with the selected queries: 
 - `original:"*20020312-190154-Cab-2002*"`
 - `original:"demo/20020312-190154-Cab-2002*"`
 - `original:"demo/20020312-190154-Cab-2002.jpg*"`
 - `original:"demo/20020312-190154-Cab-2002.jpg"`

#### 2. How can we reproduce it?

1. Go to https://demo.photoprism.app/browse?view=cards&order=name&q=original%3A%22%2a20020312-190154-Cab-2002%2a%22&public=true&quality=3
2. Be noticed that "No photos found"
3. In addition, make sure that this file exists on the demo site: demo/20020312-190154-Cab-2002.jpg

Please see the attached screenshots below: 
![image](https://user-images.githubusercontent.com/2981194/179815673-d39d17b8-45c5-4bf1-af4b-c8d1ed2a918d.png)

![image](https://user-images.githubusercontent.com/2981194/179815632-bbcb4a4c-bb67-412d-970f-2b383be7f2e8.png)

![image](https://user-images.githubusercontent.com/2981194/179815986-df036397-0bc1-43da-9cf7-758b8f8cc979.png)

**Comment by graciousgrey** [comment]
You need to use the filename search filter in this case. 

https://demo.photoprism.app/browse?view=cards&order=newest&q=filename%3A%22demo%2F20020312-190154-Cab-2002.jpg%2a%22&public=true&quality=3

The original search filter is only applicable for photos that have been imported, the cab photo has been indexed, not imported --> https://docs.photoprism.app/user-guide/library/#importing-files

**Comment by wizardion** [comment]
so, it is a little confusing that the filter `original` doesn't really search the original files, but only imported... 
The counterquestion is, how can I perform searching the file being known only the original relevant path `demo/file_name.heic`?

**Reply by graciousgrey** [reply]
In this case original stands for original name, not original file. If you import files, the original name is saved as file property and the file get's a new filename. If files only get indexed the filename does not change, so there is no original filename property.

> The counterquestion is, how can I perform searching the file being known only the original relevant path demo/file_name.heic?

Heic files are always stacked with a sidecar JPG, that can be displayed by the browser. This JPG is the primary file. For performance reasons the search currently only returns primary files. This might be improved later.

At the moment searching for `filename:"demo/file_name*"` or `filename:"demo/file_name.heic*"` should work.

**Reply by wizardion** [reply]
That makes sense, totally! 
Thank you for your quick and reliable response!

And the last question, I know that duplicated photos feature is not yet implemented but, what if the indexer marks a photo as duplicate and skips it? I guess in this case I won't be able to find a photo even if I use the API request like `/api/v1/photos?q=filename:...`, is that correct?

**Reply by graciousgrey** [reply]
You are right the indexer does not index duplicates, so you can't use the search to find those. Until https://github.com/photoprism/photoprism/issues/1308 is implemented you need to look for the filename in the duplicates database table.
